lib/ansible/modules/network/aci/mso_schema_site_anp_epg_domain.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Dag Wieers (@dagwieers) <dag@wieers.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: mso_schema_site_anp_epg_domain
+short_description: Manage site-local EPG domains in schema template
+description:
+- Manage site-local EPG domains in schema template on Cisco ACI Multi-Site.
+author:
+- Dag Wieers (@dagwieers)
+version_added: '2.8'
+options:
+  schema:
+    description:
+    - The name of the schema.
+    type: str
+    required: yes
+  site:
+    description:
+    - The name of the site.
+    type: str
+    required: yes
+  template:
+    description:
+    - The name of the template.
+    type: str
+    required: yes
+  anp:
+    description:
+    - The name of the ANP.
+    type: str
+  epg:
+    description:
+    - The name of the EPG.
+    type: str
+  domain_association_type:
+    description:
+    - The type of domain to associate.
+    type: str
+    choices: [ vmmDomain, l3ExtDomain, l2ExtDomain, physicalDomain, fibreChannel ]
+  domain_profile:
+    description:
+    - The domain profile name.
+    type: str
```

Comment:
If VMM domain, add support for 
"vmmDomainProperties": {
                    "microSegVlan": {
                      "vlanType": {},
                      "vlan": 0
                    },
                    "portEncapVlan": {
                      "vlanType": {},
                      "vlan": 0
                    },
                    "vlanEncapMode": {},
                    "allowMicroSegmentation": 0,
                    "switchType": {},
                    "switchingMode": {},
                    "epgLagPol": {
                      "enhancedLagPol": {
                        "name": "string",
                        "dn": "string"
                      }
                    }